app/controllers/project_samples_controller.rb

```diff
@@ -0,0 +1,34 @@
+class ProjectSamplesController < ApplicationController
```

Comment:
why do you need a separate controller for this? why don't you just put this under samples? 
I think you can move the current samples#index controller to some where else and just use samples#index instead of creating a new one.